engine/src/test/java/io/camunda/zeebe/engine/processing/signal/SignalIntermediateThrowEventTest.java

```diff
@@ -169,4 +171,47 @@ public void shouldApplyOutputMappings() {
     final var signalRecord = RecordingExporter.signalRecords(SignalIntent.BROADCASTED).getFirst();
     Assertions.assertThat(signalRecord.getValue()).hasSignalName(SIGNAL_NAME_1);
   }
+
+  @Test
+  public void shouldNotSendResponseWhenPartOfBatch() {
+    // given
+    final var process =
+        Bpmn.createExecutableProcess(PROCESS)
+            .startEvent()
+            .intermediateCatchEvent(
+                "start", b -> b.message(m -> m.name("start").zeebeCorrelationKey("=businessKey")))
+            .intermediateThrowEvent("signal", b -> b.signal(SIGNAL_NAME_1))
+            .endEvent()
+            .done();
+
+    ENGINE.deployment().withXmlResource(process).deploy();
+
+    final var processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(PROCESS)
+            .withVariables(Map.of("businessKey", "42"))
+            .create();
+
+    assertThat(
+            RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+                .withProcessInstanceKey(processInstanceKey)
+                .withElementId("start")
+                .limit(1)
+                .exists())
+        .isTrue();
+
+    // when
+    ENGINE.message().withName("start").withCorrelationKey("42").publish();
+
+    assertThat(
+            RecordingExporter.signalRecords(SignalIntent.BROADCASTED)
+                .withSignalName(SIGNAL_NAME_1)
+                .limit(1)
+                .exists())
```

Comment:
🔧 There is no need to use `.limit(1)` when calling `.exists()` because `.exists()` is a short-circuiting terminal operation. See Javadoc of `java.util.stream.Stream#findFirst()` for details, as it is used to implement `.exists()`. Feel free to add JavaDoc about this to `.exists()`.